### ISSUE TITLE
Adding a alternativ (input to) volume meter for (local) user

### ISSUE THREAD

**Issue by Elektrospy** [Frooxius/NeosPublic#915]
The current version of the volume meter is a general purpose one,
which needs the audio stream of a player / user to function and drive the output.

My old "workaround" until now was using the scale of the dashboard volume bar. 
Sadly the component for that ("VoiceIndicator") got removed with the ongoing UI overhaul (its now a missing one).

So the current only working method i know of, is by "forcefully" switching the user into a preset-up avatar to get the audio stream
and accessing the underlying attached volume meter to get the same result.

I want to avoid having a component / LogiX node that returns the users audio stream to get the same behavior,
without the avatar switching (security concerns).

We either could add an additional input type to the existing volume meter component / node
or add a distinct / alternative version which accepts a user (ref) as source 
and toggle able option / fallback to local user if no input is given.
(or just disable the output of the component / node)

**Comment by Frooxius** [comment]
I'm sorry, but this won't get added for VolumeMeter. All components in Neos generally follow the separation of concerns principle, which means the only job that VolumeMeter worries about is measuring volume of arbitrary audio source.

If you want to get audio source that represents voice of given user, that's a feature that's going to be added, I just need to make some adjustments to mitigate some security issues a bit. I can prioritize it, since it has come up several times through different issues.

If you're looking for something that works in the userspace though as dashboard volume bar was, there's a proper alternative now - AudioDeviceVolume. This one measures the volume of a given audio device. If the device index is set to -1 it will measure the volume of the default input audio device.

**Comment by shiftyscales** [comment]
Is this an issue you would consider prioritizing, @Frooxius? In conjunction with the recent additions in the form of the microphone tip, there is still a continued interest in potential applications for grabbing a user's voice, e.g. PA systems/phones/walkie talkies, dynamically setting up voice-reactive effects, etc.

**Comment by Frooxius** [comment]
Not until the security issues get resolved. This issue doesn't have much to do with audio itself, but rather the security of grabbing the streams directly.

Some of the optimizations done actually help resolve some issues (e.g. streaming VisemeData), which will allow to change the system so voice data doesn't get streamed to users outside of whisper bubble, but you can still see the user talking. However that requires some additions to the networking Streams first.

**Comment by H3BO3** [comment]
Could such a component be used with a grabbable so someone has to pick it up to expose their voice stream?

**Comment by Frooxius** [comment]
Yes you could. All you need to do is get grabbing user from the Grabbable and feed it as an input.

**Comment by H3BO3** [comment]
No, I mean, could it have that behavior natively to deal with the security concerns?

**Comment by Frooxius** [comment]
No. The purpose of this is to get voice stream based on input user. You can get the user through variety of ways, the component/node should not care how you got it.

Even if it did, it doesn't resolve the security concern either. You could have the user grab something unwittingly, save their stream and reuse. 

This needs the streams to be adjusted as I described above first, it won't be implemented before that.

**Comment by shiftyscales** [comment]
Got it. Thank you for the update/consideration. Security is always a priority in implementation. Thanks for taking the time to clarify what's blocking this issue.
